### PR TITLE
fix: Prof pic through social

### LINKED ISSUE #681: Profile Image not appearing in profile screen though in app drawer header
**Describe the bug**
Profile photo not set if logged in from social media. For instance i logged in from GitHub. The drawer header showed the photo but the profile screen didn't.

**To Reproduce**
1. Login using GitHub/social
2. Open app drawer
3. Profile image appears there but then go to View profile.

**Expected behavior**
The same image should appear there.

**Screenshots**

**Smartphone Info:**
Realme 5 Pro

**Would you like to work on the issue?**

- [x] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by jddeep**
Fixes #681

Changes: Prof pic through social

### CODE REVIEW

File: app/src/main/java/com/codingblocks/cbonlineapp/dashboard/DashboardActivity.kt [hunk 1]
```diff
@@ -310,7 +304,8 @@ class DashboardActivity : BaseCBActivity(),
             startActivity<LoginActivity>()
             finish()
         } else {
-            startActivity<ProfileActivity>()
+            val args = Pair("user_social_image", vm.prefs.SP_USER_IMAGE)
```
Comment (aggarwalpulkit596) [inline app/src/main/java/com/codingblocks/cbonlineapp/dashboard/DashboardActivity.kt]: Not to be done like this

Comment (jddeep) [reply app/src/main/java/com/codingblocks/cbonlineapp/dashboard/DashboardActivity.kt]: @aggarwalpulkit596 Then what do you suggest here?

**Comment by aggarwalpulkit596** [comment]
Use Profile View Model for this

**Comment by jddeep** [comment]
@aggarwalpulkit596 Yes, i had approached that. Can your let me know the key of the map for updating the user pic? Or if you can let me know where to find the key in our codebase

**Comment by aggarwalpulkit596** [comment]
@jddeep try to understand the workflow of the dashboard activity it's quite straight forward

**Comment by jddeep** [comment]
@aggarwalpulkit596 You are indicating to use the `updateUser` callback from the ProfileViewModel right?

**Comment by aggarwalpulkit596** [comment]
no i meant fetchUser from dashboard view model

**Comment by jddeep** [comment]
But can I update the user's pic in the fetch user callback? Are you suggesting `it.photo = <imageUrl>` inside the fetchUser callback in the Dashboard activity?

**Comment by aggarwalpulkit596** [comment]
Yes you can update that and try to understand the flow of user profile picture in dashboard same can be used for profile activity

**Comment by jddeep** [comment]
@aggarwalpulkit596 Thanks for the suggestion. Turns out was pretty straight forward indeed :)

File: app/src/main/java/com/codingblocks/cbonlineapp/dashboard/DashboardActivity.kt [hunk 2]
```diff
@@ -34,13 +34,7 @@ import com.codingblocks.cbonlineapp.tracks.LearningTracksActivity
...
 import com.codingblocks.cbonlineapp.util.JWTUtils
 import com.codingblocks.cbonlineapp.util.UNAUTHORIZED
-import com.codingblocks.cbonlineapp.util.extensions.colouriseToolbar
-import com.codingblocks.cbonlineapp.util.extensions.loadImage
-import com.codingblocks.cbonlineapp.util.extensions.observer
-import com.codingblocks.cbonlineapp.util.extensions.setToolbar
-import com.codingblocks.cbonlineapp.util.extensions.showSnackbar
-import com.codingblocks.cbonlineapp.util.extensions.slideDown
-import com.codingblocks.cbonlineapp.util.extensions.slideUp
+import com.codingblocks.cbonlineapp.util.extensions.*
```
Comment (aggarwalpulkit596) [inline app/src/main/java/com/codingblocks/cbonlineapp/dashboard/DashboardActivity.kt]: Don't use wildcards

File: app/src/main/java/com/codingblocks/cbonlineapp/dashboard/DashboardActivity.kt [hunk 3]
```diff
@@ -118,6 +112,8 @@ class DashboardActivity : BaseCBActivity(),
                 findViewById<CircleImageView>(R.id.navHeaderImageView).loadImage(vm.prefs.SP_USER_IMAGE, true)
                 findViewById<TextView>(R.id.navUsernameTv).text = ("Hello ${vm.prefs.SP_USER_NAME}")
             }
+            if (vm.prefs.SP_USER_IMAGE.isNotEmpty())
```
Comment (aggarwalpulkit596) [inline app/src/main/java/com/codingblocks/cbonlineapp/dashboard/DashboardActivity.kt]: Not needed

File: app/src/main/java/com/codingblocks/cbonlineapp/profile/ProfileActivity.kt [hunk 4]
```diff
@@ -60,7 +56,7 @@ class ProfileActivity : BaseCBActivity() {
             }
             nameTv.text = "${it.firstname} ${it.lastname}"
             userNameTv.text = it.username
-            userImgView.loadSvg(it.photo ?: "")
+            userImgView.loadImage(it.photo ?: "", scale = true)
```
Comment (aggarwalpulkit596) [inline app/src/main/java/com/codingblocks/cbonlineapp/profile/ProfileActivity.kt:63]: Yup only this was to be done any guesses why ?

File: app/src/main/java/com/codingblocks/cbonlineapp/profile/ProfileActivity.kt [hunk 5]
```diff
@@ -13,11 +13,7 @@ import com.codingblocks.cbonlineapp.database.AppDatabase
 import com.codingblocks.cbonlineapp.util.FileUtils
 import com.codingblocks.cbonlineapp.util.JWTUtils
 import com.codingblocks.cbonlineapp.util.PreferenceHelper
-import com.codingblocks.cbonlineapp.util.extensions.loadSvg
-import com.codingblocks.cbonlineapp.util.extensions.observer
-import com.codingblocks.cbonlineapp.util.extensions.setToolbar
-import com.codingblocks.cbonlineapp.util.extensions.showDialog
-import com.codingblocks.cbonlineapp.util.extensions.showSnackbar
+import com.codingblocks.cbonlineapp.util.extensions.*
```
Comment (aggarwalpulkit596) [inline app/src/main/java/com/codingblocks/cbonlineapp/profile/ProfileActivity.kt]: Same

**Review by aggarwalpulkit596 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/codingblocks/cbonlineapp/profile/ProfileActivity.kt (hunk 4 above)
Comment (jddeep) [reply app/src/main/java/com/codingblocks/cbonlineapp/profile/ProfileActivity.kt:63]: Yeah that's am img url mostly jpg, can't load it as svg

File: app/src/main/java/com/codingblocks/cbonlineapp/profile/ProfileActivity.kt (hunk 5 above)
Comment (jddeep) [reply app/src/main/java/com/codingblocks/cbonlineapp/profile/ProfileActivity.kt]: Okay am doing it asap

File: app/src/main/java/com/codingblocks/cbonlineapp/dashboard/DashboardActivity.kt (hunk 3 above)
Comment (jddeep) [reply app/src/main/java/com/codingblocks/cbonlineapp/dashboard/DashboardActivity.kt]: Aight

**Comment by aggarwalpulkit596** [comment]
Also build js failing PTAL at that

**Comment by jddeep** [comment]
@aggarwalpulkit596 Can't quite understand the reason for the build failures :thinking:

**Review by aggarwalpulkit596 — APPROVED** [review]

File: app/src/main/java/com/codingblocks/cbonlineapp/dashboard/DashboardActivity.kt [hunk 6]
```diff
@@ -119,6 +119,7 @@ class DashboardActivity : BaseCBActivity(),
                 findViewById<CircleImageView>(R.id.navHeaderImageView).loadImage(vm.prefs.SP_USER_IMAGE, true)
                 findViewById<TextView>(R.id.navUsernameTv).text = ("Hello ${vm.prefs.SP_USER_NAME}")
             }
+            it.photo = vm.prefs.SP_USER_IMAGE
```
Comment (aggarwalpulkit596) [inline app/src/main/java/com/codingblocks/cbonlineapp/dashboard/DashboardActivity.kt:122]: not needed

File: onlineapi/src/main/kotlin/com/codingblocks/onlineapi/models/APIModel.kt [hunk 7]
```diff
@@ -321,7 +321,7 @@ data class Course(
         val lastReadNotification: String?,
         val lastname: String,
         val oneauthId: String?,
-        val photo: String?,
+        var photo: String?,
```
Comment (aggarwalpulkit596) [inline onlineapi/src/main/kotlin/com/codingblocks/onlineapi/models/APIModel.kt]: not needed

File: app/src/main/java/com/codingblocks/cbonlineapp/dashboard/DashboardActivity.kt (hunk 6 above)
Comment (aggarwalpulkit596) [inline app/src/main/java/com/codingblocks/cbonlineapp/dashboard/DashboardActivity.kt]: not needed

**Review by aggarwalpulkit596 — CHANGES_REQUESTED** [review]

**Comment by jddeep** [comment]
@aggarwalpulkit596 done

**Review by aggarwalpulkit596 — APPROVED** [review]

**Pull request merged**